test/core/event_engine/posix/posix_engine_listener_utils_test.cc

```diff
@@ -0,0 +1,162 @@
+  EXPECT_TRUE(result.ok());
+  EXPECT_GT(*result, 0);
+  port = *result;
+  EXPECT_GE(listener_sockets.Size(), 1);
+  EXPECT_LE(listener_sockets.Size(), 2);
+  for (auto socket = listener_sockets.begin(); socket != listener_sockets.end();
+       ++socket) {
+    ASSERT_TRUE((*socket).addr.address()->sa_family == AF_INET6 ||
+                (*socket).addr.address()->sa_family == AF_INET);
+    if ((*socket).addr.address()->sa_family == AF_INET6) {
+      EXPECT_EQ(SockaddrToString(&(*socket).addr, true).value(),
+                absl::StrCat("[::]:", std::to_string(port)));
+    } else if ((*socket).addr.address()->sa_family == AF_INET) {
+      EXPECT_EQ(SockaddrToString(&(*socket).addr, true).value(),
+                absl::StrCat("0.0.0.0:", std::to_string(port)));
+    }
+    close(socket->sock.Fd());
+  }
+}
+
+#ifdef GRPC_HAVE_IFADDRS
+TEST(PosixEngineListenerUtils, ListenerContainerAddAllLocalAddressesTest) {
+  TestListenerSocketsContainer listener_sockets;
+  int port = grpc_pick_unused_port_or_die();
+  ChannelArgsEndpointConfig config;
+  struct ifaddrs* ifa = nullptr;
+  struct ifaddrs* ifa_it;
+  if (getifaddrs(&ifa) != 0 || ifa == nullptr) {
+    // No ifaddresses available.
+    gpr_log(GPR_INFO,
+            "Skipping ListenerAddAllLocalAddressesTest because the machine "
+            "does not have interfaces configured for listening.");
+    return;
+  }
+  int num_ifaddrs = 0;
+  for (ifa_it = ifa; ifa_it != nullptr; ifa_it = ifa_it->ifa_next) {
+    ++num_ifaddrs;
+  }
+  freeifaddrs(ifa);
+  auto result = ListenerContainerAddAllLocalAddresses(
+      listener_sockets, TcpOptionsFromEndpointConfig(config), port);
+  if (num_ifaddrs == 0 || !result.ok()) {
+    // Its possible that the machine may not have any Ipv4/Ipv6 interfaces
+    // configured for listening. In that case, dont fail test.
+    gpr_log(GPR_INFO,
+            "Skipping ListenerAddAllLocalAddressesTest because the machine "
+            "does not have Ipv6/Ipv6 interfaces configured for listening.");
+    return;
+  }
+  // Some sockets have been created and bound to interfaces on the machiene.
+  // Verify that they are listening on the correct port.
+  EXPECT_GT(*result, 0);
+  port = *result;
+  EXPECT_GE(listener_sockets.Size(), 1);
+  EXPECT_LE(listener_sockets.Size(), num_ifaddrs);
+  for (auto socket = listener_sockets.begin(); socket != listener_sockets.end();
+       ++socket) {
+    ASSERT_TRUE((*socket).addr.address()->sa_family == AF_INET6 ||
+                (*socket).addr.address()->sa_family == AF_INET);
+    EXPECT_EQ(SockaddrGetPort((*socket).addr), port);
+    close(socket->sock.Fd());
+  }
+}
+#endif  // GRPC_HAVE_IFADDRS
+
+}  // namespace posix_engine
+}  // namespace grpc_event_engine
+
+int main(int argc, char** argv) {
+  ::testing::InitGoogleTest(&argc, argv);
+  return RUN_ALL_TESTS();
+}
+
+#else /* GRPC_POSIX_SOCKET_UTILS_COMMON */
+
+int main(int argc, char** argv) { return 1; }
```

Comment:
The intent is to fail the test if socket utils are unsupported? I'm curious when this bit of code would be executed.

Reply:
Good catch. No the intent is to not fail. I changed it to return 0. But I'm not sure which platforms (if any) may trigger this code to be executed